tmo-monitor.py

```diff
@@ -134,79 +150,217 @@ def sha256url(self, val1, val2):
     return self.base64url_escape(self.sha256(val1, val2))
 
 
+class Configuration:
+  def __init__(self):
+    # Set default values
+    self.reboot_now = False
+    self.skip_reboot = False
+    self.login = dict([('username', 'admin'), ('password', '')])
+    self.ping = dict([('interface', ''), ('ping_host', 'google.com'), ('ping_count', 1), ('ping_interval', 10)])
+    self.connection = dict([('primary_band', ''), ('secondary_band', 'n41'), ('enbid', '')])
+    self.reboot = dict([('uptime', 90), ('ping', False), ('4G_band', False), ('5G_band', False), ('enbid', False)])
+    self.general = dict([('print_config', False)])
+  
+    # Command line arguments override defaults & .env file
+    self.read_environment()
+    args = self.parse_commandline()
+    self.parse_arguments(args)
+
+    if self.skip_reboot and self.reboot_now:
+      print('Incompatible options: --reboot and --skip-reboot\n', file=sys.stderr)
+      self.parser.print_help(sys.stderr)
+      sys.exit(2)
+    if self.skip_reboot:
+      for var in {'ping', '4G_band', '5G_band', 'enbid'}:
+        self.reboot[var] = False
+    # if not self.login['username']:
+    #   print('The following argument is required: username')
+    #   sys.exit(2)
+    if not self.login['password']:    
+      self.password = getpass.getpass('Password: ')
+
+
+  def read_environment(self):
+    try:
+      envfile=find_dotenv()
+      load_dotenv(envfile)
+    except:
+      logging.debug("No .env file found")
+      return
+    for var in {'username', 'password'}:
+      tmp = os.environ.get('tmo_' + var)
+      if tmp != None:
+        self.login[var] = tmp
+    for var in {'interface', 'ping_host', 'ping_count', 'ping_interval'}:
+      tmp = os.environ.get('tmo_' + var)
+      if tmp != None:
+        self.ping[var] = tmp  
+    for var in {'primary_band', 'secondary_band'}:
+      tmp = os.environ.get('tmo_' + var)
+      if tmp != None:
+        splits = tmp.split(',')
+        self.connection[var] = splits
+    tmp = os.environ.get('tmo_enbid')
+    if tmp != None:
+      self.connection['enbid'] = tmp
+    for var in {'uptime', 'ping', '4G_band', '5G_band', 'enbid'}:
+      tmp = os.environ.get('tmo_' + var + '_reboot')
+      if tmp != None:
+        if tmp.lower() == 'true':
+          self.reboot[var] = True
+        else:
+          self.reboot[var] = False
+    tmp = os.environ.get('tmo_skip_reboot')
+    if tmp != None:
+      self.skip_reboot = tmp
```

Comment:
seems it makes more sense to set this directly to `True` rather than relying on this being a truthy value

Reply:
Fixed, thanks.